packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m

```diff
@@ -675,99 +701,117 @@ - (NSDictionary *)encodingMap
 
 #pragma mark - Internal methods
 
-- (void)_downloadResumableCreateSessionWithUrl:(NSURL *)url withScopedPath:(NSString *)scopedPath withUUID:(NSString *)uuid withOptions:(NSDictionary *)options withResumeData:(NSData * _Nullable)resumeData withResolver:(UMPromiseResolveBlock)resolve withRejecter:(UMPromiseRejectBlock)reject
+- (NSMutableURLRequest *)_createRequest:(NSURL *)url headers:(NSDictionary * _Nullable)headers
 {
-  NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
-  sessionConfiguration.requestCachePolicy = NSURLRequestReloadIgnoringLocalCacheData;
-  sessionConfiguration.URLCache = nil;
-  
-  __weak typeof(self) weakSelf = self;
-  EXDownloadDelegateOnWriteCallback onWrite = ^(NSURLSessionDownloadTask *task, int64_t bytesWritten, int64_t totalBytesWritten, int64_t totalBytesExpectedToWrite) {
-    __strong EXFileSystem *strongSelf = weakSelf;
-    if (strongSelf && bytesWritten > 0) {
-      [strongSelf sendEventWithName:EXDownloadProgressEventName
-                               body:@{@"uuid":uuid,
-                                      @"data":@{
-                                          @"totalBytesWritten": @(totalBytesWritten),
-                                          @"totalBytesExpectedToWrite": @(totalBytesExpectedToWrite),
-                                          },
-                                      }];
+  NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:url];
+  if (headers != nil) {
+    for (NSString *headerKey in headers) {
+      [request setValue:[headers valueForKey:headerKey] forHTTPHeaderField:headerKey];
     }
-  };
+  }
   
-  EXDownloadDelegateOnDownloadCallback onDownload = ^(NSURLSessionDownloadTask *task, NSURL *location) {
-    NSURL *scopedLocation = [NSURL fileURLWithPath:scopedPath];
-    NSData *locationData = [NSData dataWithContentsOfURL:location];
-    [locationData writeToFile:scopedPath atomically:YES];
-    NSData *data = [NSData dataWithContentsOfURL:scopedLocation];
-    if (!data) {
-      reject(@"E_UNABLE_TO_SAVE",
-             @"Unable to save file to local URI",
-             nil);
-      return;
-    }
-    NSMutableDictionary *result = [NSMutableDictionary dictionary];
-    result[@"uri"] = scopedLocation.absoluteString;
-    result[@"complete"] = @(YES);
-    if (options[@"md5"]) {
-      result[@"md5"] = [data md5String];
-    }
-    NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)task.response;
-    result[@"status"] = @([httpResponse statusCode]);
-    result[@"headers"] = [httpResponse allHeaderFields];
-    
-    __strong EXFileSystem *strongSelf = weakSelf;
-    if (strongSelf) {
-      [strongSelf.downloadObjects removeObjectForKey:uuid];
+  return request;
+}
+
+- (EXFileSystemSessionType)_importSessionType:(NSNumber * _Nullable)type
+{
+  return [type intValue] ?: EXFileSystemBackgroundSession;
```

Comment:
This `?:` seems invalid — it expands to:
```objc
if (![type intValue]) {
  return [type intValue];
} else {
  return EXFileSystemBackgroundSession;
}
```
So we will use fallback not only if `type` is `nil` (`[nil intValue]` returns `0`, 👍), but also if `type == @(0)`, which coincidentally would return the same value (`EXFileSystemBackgroundSession == 0`), but it looks strange.

Would you mind rephrasing the code so it's clear it doesn't contain a bug?